hyperon_das/query_engines/local_query_engine.py

```diff
@@ -49,7 +49,7 @@ def _recursive_query(
         self,
         query: Query,
         mappings: Set[Assignment] | None = None,  # TODO: this parameter is never used
-        parameters: Dict[str, Any] = [],
+        parameters: dict[str, Any] | None = None,
```

Comment:
Seems like the use of dict is not standardized, sometime we are using Dict, with a capital `D`, other times dict with a lowercase `d`.
My personal preference is to use the lowercase across the board, so we don't have to add the import to it.

Reply:
We plan to move to Python's default types for typing, but for now, this will only apply to new or rewritten features.